src/regexp/ast_to_re.cc

```diff
@@ -251,7 +253,7 @@ RE *ast_to_re(RESpec &spec, const AST *ast, size_t &ncap, int32_t height)
 
         case AST::REF:
             if (misuse_of_named_def(spec, ast)) {
-                return NULL;
+                exit(1);
```

Comment:
Don't change this, some day `misuse_of_named_def` will return a proper error, and the caller will need to check it and return up the stack. If you want to clean it up for now, replace `misuse_of_named_def` with a void function `check_misuse_of_named_def` and remove the `if`.

Reply:
Well, this is how re2c works right now. Do you remember, `fatal` calls `exit(1)`?

Reply:
Added here and below

```c
                /* unreachable until 'misuse_of_named_def' uses 'exit(1)' */
                return NULL;
```

Reply:
I think it's better to change it like this (pseudo-diff):
```diff
- if (misuse_of_named_def(spec, ast)) return NULL;
+ check_misuse_of_named_def(spec, ast);
```
And the function definition:
```diff
- bool misuse_of_named_def(RESpec &spec, const AST *ast)
+ void misuse_of_named_def(RESpec &spec, const AST *ast)
```
Having a comment is way too much hassle for such a minor thing, the code is already complex by itself.

Reply:
I changed this one:

```diff
diff --git a/src/regexp/ast_to_re.cc b/src/regexp/ast_to_re.cc
index f0dd9cea..9017dbdb 100644
--- a/src/regexp/ast_to_re.cc
+++ b/src/regexp/ast_to_re.cc
@@ -60,7 +60,7 @@ static Range *char_to_range(RESpec &, const ASTChar &, bool);
 static Range *diff_to_range(RESpec &, const AST *);
 static Range *dot_to_range(RESpec &, const AST *);
 static Range *cls_to_range(RESpec &, const AST *);
-static bool misuse_of_named_def(RESpec &, const AST *);
+static void check_misuse_of_named_def(RESpec &, const AST *);
 static void assert_tags_used_once(RESpec &, const Rule &, const std::vector<Tag> &);
 static void init_rule(RESpec &, Rule &, const SemAct *, const std::vector<Tag> &, size_t, size_t);
 static bool is_icase(const opt_t *, bool);
@@ -252,10 +252,7 @@ RE *ast_to_re(RESpec &spec, const AST *ast, size_t &ncap, int32_t height)
             }

         case AST::REF:
-            if (misuse_of_named_def(spec, ast)) {
-                /* unreachable until 'misuse_of_named_def' uses 'exit(1)' */
-                return NULL;
-            }
+            check_misuse_of_named_def(spec, ast);
             return ast_to_re(spec, ast->ref.ast, ncap, height);

         case AST::ITER: {
@@ -368,11 +365,8 @@ Range *ast_to_range(RESpec &spec, const AST *ast)
             if (spec.opts->posix_syntax) break;
             return ast_to_range(spec, ast->cap);
         case AST::REF:
-            if (misuse_of_named_def(spec, ast)) {
-                /* unreachable until 'misuse_of_named_def' uses 'exit(1)' */
-                return NULL;
-            }
-          return ast_to_range(spec, ast->ref.ast);
+            check_misuse_of_named_def(spec, ast);
+            return ast_to_range(spec, ast->ref.ast);
         case AST::CLS:
             return cls_to_range(spec, ast);
         case AST::DOT:
@@ -444,7 +438,7 @@ RE *re_class(RESpec &spec, const loc_t &loc, const Range *r)
     return NULL; /* unreachable */
 }

-bool misuse_of_named_def(RESpec &spec, const AST *ast)
+void check_misuse_of_named_def(RESpec &spec, const AST *ast)
 {
     DASSERT(ast->type == AST::REF);

@@ -455,7 +449,6 @@ bool misuse_of_named_def(RESpec &spec, const AST *ast)
             , ast->ref.name->c_str());
         exit(1);
     }
-    return false;
 }

 void assert_tags_used_once(RESpec &spec, const Rule &rule
```